src/pages/components/Overlay.qml

```diff
@@ -486,6 +487,27 @@ Background {
                         pageStack.animatorPush("com.jolla.settings.system.CertificateDetailsPage", {"website": webView.security.subjectDisplayName, "details": webView.security.serverCertDetails})
                     }
                 }
+                onSavePageAsPDF: {
+                    var filename = "unnamed_file"
+                    if (webView.title && webView.title.length !== 0) {
+                        filename = webView.title
+                    } else {
+                        // drop query string from URL and split to sections
+                        var urlSections = webView.url.split("?")[0].split("/")
```

Comment:
WebUtils.displayableUrl(url) does conceptually the same. It's ok to move that WebUtils.displayableUrl to the declarativewebpage as a property and refer here as "webView.contentItem.displayableUrl"

Reply:
WebUtils.displayableUrl(url) returns host instead of filename. For example, if pass http://xxxx.xxx/aaa/bbb.html?zzz, it will return xxxx.xxx instead of bbb.html.